src/main/java/net/dv8tion/jda/core/events/message/react/MessageReactionRemoveAllEvent.java

```diff
@@ -16,31 +16,41 @@
 
 package net.dv8tion.jda.core.events.message.react;
 
+import net.dv8tion.jda.client.entities.Group;
 import net.dv8tion.jda.core.JDA;
+import net.dv8tion.jda.core.entities.Guild;
 import net.dv8tion.jda.core.entities.MessageChannel;
-import net.dv8tion.jda.core.events.Event;
+import net.dv8tion.jda.core.entities.PrivateChannel;
+import net.dv8tion.jda.core.entities.TextChannel;
+import net.dv8tion.jda.core.events.message.GenericMessageEvent;
 
-public class MessageReactionRemoveAllEvent extends Event
+public class MessageReactionRemoveAllEvent extends GenericMessageEvent
 {
 
-    protected String messageId;
-    protected MessageChannel channel;
+    public MessageReactionRemoveAllEvent(JDA api, long responseNumber, long messageId, MessageChannel channel)
+    {
+        super(api, responseNumber, messageId, channel);
+    }
+
+    public Guild getGuild()
+    {
+        TextChannel channel = getTextChannel();
+        return channel != null ? channel.getGuild() : null;
+    }
 
-    public MessageReactionRemoveAllEvent(JDA api, long responseNumber, String messageId, MessageChannel channel)
+    public TextChannel getTextChannel()
     {
-        super(api, responseNumber);
-        this.messageId = messageId;
-        this.channel = channel;
+        return getChannel() instanceof TextChannel ? (TextChannel) getChannel() : null;
     }
 
-    public String getMessageId()
+    public PrivateChannel getPrivateChannel()
     {
-        return messageId;
+        return getChannel() instanceof PrivateChannel ? (PrivateChannel) getChannel() : null;
     }
 
-    public MessageChannel getChannel()
+    public Group getGroup()
     {
-        return channel;
+        return getChannel() instanceof Group ? (Group) getChannel() : null;
```

Comment:
These should be reworked to use ChannelType instead of 2 calls to `getChannel()`